exchanges/binance/binance_websocket.go

```diff
@@ -94,142 +133,270 @@ func (b *Binance) WsHandleData() {
+	err := json.Unmarshal(respRaw, &multiStreamData)
+	if err != nil {
+		return err
+	}
+	if method, ok := multiStreamData["method"].(string); ok {
+		if strings.EqualFold(method, "subscribe") {
+			return nil
+		}
+		if strings.EqualFold(method, "unsubscribe") {
+			return nil
+		}
+	}
+	if e, ok := multiStreamData["e"].(string); ok {
+		switch e {
+		case "outboundAccountInfo":
+			var data wsAccountInfo
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to outboundAccountInfo structure %s",
+					b.Name,
+					err)
+			}
+			b.Websocket.DataHandler <- data
+		case "outboundAccountPosition":
+			var data wsAccountPosition
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to outboundAccountPosition structure %s",
+					b.Name,
+					err)
+			}
+			b.Websocket.DataHandler <- data
+		case "balanceUpdate":
+			var data wsBalanceUpdate
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to balanceUpdate structure %s",
+					b.Name,
+					err)
+			}
+			b.Websocket.DataHandler <- data
+		case "executionReport":
+			var data wsOrderUpdate
+			err := json.Unmarshal(respRaw, &data)
+			if err != nil {
+				return fmt.Errorf("%v - Could not convert to executionReport structure %s",
+					b.Name,
+					err)
+			}
+			var orderID = strconv.FormatInt(data.OrderID, 10)
+			oType, err := order.StringToOrderType(data.OrderType)
+			if err != nil {
+				b.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: b.Name,
+					OrderID:  orderID,
+					Err:      err,
 				}
-
-				amount, err := strconv.ParseFloat(trade.Quantity, 64)
-				if err != nil {
-					b.Websocket.DataHandler <- fmt.Errorf("%v - amount conversion error: %s",
-						b.Name,
-						err)
-					continue
+			}
+			var oSide order.Side
+			oSide, err = order.StringToOrderSide(data.Side)
+			if err != nil {
+				b.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: b.Name,
+					OrderID:  orderID,
+					Err:      err,
 				}
-
-				b.Websocket.DataHandler <- wshandler.TradeData{
-					CurrencyPair: currency.NewPairFromFormattedPairs(trade.Symbol, b.GetEnabledPairs(asset.Spot),
-						b.GetPairFormat(asset.Spot, true)),
-					Timestamp: time.Unix(0, trade.TimeStamp*int64(time.Millisecond)),
-					Price:     price,
-					Amount:    amount,
-					Exchange:  b.Name,
-					AssetType: asset.Spot,
-					Side:      trade.EventType,
+			}
+			var oStatus order.Status
+			oStatus, err = stringToOrderStatus(data.CurrentExecutionType)
+			if err != nil {
+				b.Websocket.DataHandler <- order.ClassificationError{
+					Exchange: b.Name,
+					OrderID:  orderID,
+					Err:      err,
 				}
-				continue
-			case "ticker":
-				t := TickerStream{}
-				err := json.Unmarshal(multiStreamData.Data, &t)
+			}
+			var p currency.Pair
+			var a asset.Item
+			p, a, err = b.GetRequestFormattedPairAndAssetType(data.Symbol)
```

Comment:
We'll need to address this in another PR as this logic will break down when an exchange has spot and perpetual swaps (since they can both be BTC-USD, but would be fine for indexes and futures (but dependent on each exchange).) currency.NewPairFromFormattedPairs is less error prone since you have to specify the asset type

Reply:
Should also mention that a better solution for this would be to detect the context in the exchange wrapper or func, then use the specific asset type, but this outside the scope of this PR